src/main/java/fact/TriggerEmulation/EmulateDiscriminator.java

```diff
@@ -0,0 +1,120 @@
+package fact.TriggerEmulation;
+
+import fact.Constants;
+import fact.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import stream.Data;
+import stream.Processor;
+import stream.annotations.Parameter;
+
+import static fact.TriggerEmulation.Discriminator.booleanToInt;
+import static fact.TriggerEmulation.Discriminator.discriminatePatches;
+
+/**
+ * Emulate a discriminator that is working on the summed timeseries of the patches. Signals are digitized according to
+ * the provided threshold and the minimum time a signal has to stay above it.
+ * Created by jbuss on 15.11.17.
+ */
+public class EmulateDiscriminator implements Processor{
+
+    static Logger log = LoggerFactory.getLogger(EmulateDiscriminator.class);
+
+    @Parameter(required = true)
+    public String key;
+
+    @Parameter(required = false,
+            description = "boolean array [number of patches] flagging if patch triggered ")
+    public String primitivesKey = "TriggerPrimitives";
+
+    @Parameter(required = false,
+            description = "int array [number of patches] containing each the first slice above threshold ")
+    public String triggerSliceKey = "TriggerSlice";
+
+    @Parameter(required = false,
+            description = "Converts the patch array into a 1440*ROI array inorder to visualize the patche time series in the viewer")
+    public Boolean visualize = false;
+
+    @Parameter(required = false,
+            description = "threshold of the discriminator in DAC units")
+    public int threshold = 330;
+
+    @Parameter(required = false,
+            description = "key in dataitem with threshold of the discriminator in DAC units. !!!!Overwrites threshold")
+    public String thresholdKey = null;
+
+    @Parameter(required = false,
+            description = "minimum time the signal has to stay above the threhold")
+    public int minTimeOverThreshold = 8;
+
+    @Parameter(required = false,
+            description = "number of slices to ignore at the beginning of the time series")
+    public int skipFirst = 10;
+
+    @Parameter(required = false,
+            description = "number of slices to ignore at the end of the time series")
+    public int skipLast = 40;
+
+
+    public int default_slice = Integer.MAX_VALUE;
+
+    @Override
+    public Data process(Data item) {
+        double[][] data = (double[][]) item.get(key);
+
+        int n_patches = Constants.N_PATCHES;
+        double millivoltPerDAC = Constants.MILLIVOLT_PER_DAC;
+
+        if (thresholdKey != null){
+            Utils.isKeyValid(item, thresholdKey, Integer.class);
+            threshold = (Integer) item.get(thresholdKey);
+        }
+
+
+        int[] patchTriggerSlice = new int[n_patches];
+
+        boolean[] triggerPrimitives = discriminatePatches(
+                data,
+                n_patches,
+                patchTriggerSlice,
+                threshold,
+                minTimeOverThreshold,
+                skipFirst,
+                skipLast
```

Comment:
I believe the 4-tuple consisting of `threshold, minTimeOverThreshold, skipFirst, skipLast` forms some kind of a group in large portions of this code, so maybe one could tidy up by creating a little class/struct for these 4?

Reply:
Not sure if this makes sense. In the end it is just four variables each being an int. However, `skipFirst` and `skipLast` are used in many different processors in order to define a valid operation window on the time series. Maybe we should open an issue to add a class `ValidTimeSeriesWindow` and refactor all processors using a valid window. I think it would even make sense to serialize this class and hand it to the data item so other processors working with a window can used hat specific window too. What do you think.